### ISSUE TITLE
CA2247 should flag wrong arguments via base

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn-analyzers#5135]
### Analyzer

**Diagnostic ID**: [CA2247](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2247): `Argument passed to TaskCompletionSource constructor should be TaskCreationOptions enum instead of TaskContinuationOptions enum`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug

The analyzer fails to see incorrect arguments when passed to the ctor via a derived class' use of base.

### Steps To Reproduce

```C#
class Derived : TaskCompletionSource<int>
{
    public Derived() : base(TaskContinuationOptions.RunContinuationsAsynchronously) { }
}
```

### Expected behavior

Should fire CA2247 on the argument to `base(...)`

### Actual behavior

CA2247 isn't raised.

**Comment by jmarolf** [comment]
@jeffhandley can someone take a look here?

**Comment by jeffhandley** [comment]
@buyaa-n / @carlossanlop - Can you triage this to determine if it's an easy fix?

**Comment by stephentoub** [comment]
> Can you triage this to determine if it's an easy fix?

There is:
https://github.com/dotnet/roslyn-analyzers/pull/5150
